win32ss/reactx/ntddraw/dxeng.c

```diff
@@ -463,7 +464,8 @@ DxEngGetDCState(HDC hDC,
                 retVal = (DWORD_PTR) pDC->fs & DC_FLAG_FULLSCREEN;
                 break;
             case 2:
-                UNIMPLEMENTED;
+                /* Return the number of rectangles in the visible region. */
+                retVal = (DWORD_PTR) pDC->prgnVis->rdh.nCount;
```

Comment:
Should that be:
retVal = (DWORD_PTR) pDC->prgnRao ? pDC->prgnRao->rdh.nCount : pDC->prgnVis->rdh.nCount;

Reply:
Hooray! :tada: After applying your suggestion locally, the BSOD mentioned in that ticket does no longer happen for me when trying run DirectDraw test or any DDraw games with replaced files! :smiley: (at least in VirtualBox 5.1.38, and even the 1st DirectDraw test succeeds!) So yes, it definitely should. Thanks @jimtabor! :+1: 
Although the DDraw test/games don't work anyway, but nevertheless the main problem is now fixed.

Reply:
Vert Good!